io-engine/src/eventing/host_events.rs

```diff
@@ -0,0 +1,46 @@
+use events_api::event::{
+    EventAction,
+    EventCategory,
+    EventMessage,
+    EventMeta,
+    EventSource,
+};
+
+use crate::{
+    core::MayastorEnvironment,
+    eventing::{EventMetaGen, EventWithMeta},
+    subsys::NvmfSubsystem,
+};
+use spdk_rs::NvmfController;
+
+impl EventMetaGen for NvmfSubsystem {
+    fn meta(&self) -> EventMeta {
+        let event_source = EventSource::new(
+            MayastorEnvironment::global_or_default().node_name,
+        )
+        .with_subsystem_data(&self.get_nqn());
+
+        EventMeta::from_source(event_source)
+    }
+}
+
+impl EventWithMeta for NvmfController {
+    fn event(
+        &self,
+        event_action: EventAction,
+        meta: EventMeta,
+    ) -> EventMessage {
+        let mut event_meta = meta.clone();
```

Comment:
Do we need to clone, can't we just mutate meta?

Reply:
Yes we can mutate meta. Added the change now.